File: src/main/java/com/offz/spigot/custommobs/Pathfinders/PathfinderGoalTemptPitchLock.java
package com.offz.spigot.custommobs.Pathfinders;

import net.minecraft.server.v1_13_R2.*;
import org.bukkit.craftbukkit.v1_13_R2.entity.CraftLivingEntity;
import org.bukkit.craftbukkit.v1_13_R2.event.CraftEventFactory;
import org.bukkit.event.entity.EntityTargetEvent;
import org.bukkit.event.entity.EntityTargetLivingEntityEvent;

public class PathfinderGoalTemptPitchLock extends PathfinderGoal {
    private final EntityCreature a;
    private final double b;
    private final Item[] targetItems;
    private final boolean l;
    private double c;
    private double d;
    private double e;
    private double f;
    private double g;
    private EntityLiving target;
    private int i;
    private boolean j;

    public PathfinderGoalTemptPitchLock(EntityCreature entitycreature, double d0, Item[] items, boolean flag) {
        this(entitycreature, d0, flag, items);
    }

    public PathfinderGoalTemptPitchLock(EntityCreature entitycreature, double d0, boolean flag, Item[] items) {
        this.a = entitycreature;
        this.b = d0;
        this.targetItems = items;
        this.l = flag;
        this.a(3);
        if (!(entitycreature.getNavigation() instanceof Navigation)) {
            throw new IllegalArgumentException("Unsupported mob type for TemptGoal");
        }
    }

    public boolean a() {
        if (this.i > 0) {
            --this.i;
            return false;
        } else {
            this.target = this.a.world.findNearbyPlayer(this.a, 10.0D);
            boolean tempt = false;
            if (this.target != null)
                for (Item item : this.targetItems)
                    if (this.target.getItemInMainHand().getItem().equals(item)) {
                        tempt = true;
                        break;
                    }
//                    tempt = true = this.target == null ? false : this.k.choices(this.target.getItemInMainHand()) || this.a(this.target.getItemInOffHand());
            if (tempt) {
                EntityTargetLivingEntityEvent event = CraftEventFactory.callEntityTargetLivingEvent(this.a, this.target, EntityTargetEvent.TargetReason.TEMPT);
                if (event.isCancelled()) {
                    return false;
                }

                this.target = event.getTarget() == null ? null : ((CraftLivingEntity) event.getTarget()).getHandle();
            }

            return tempt;
        }
    }

    public void e() {
        this.a.getControllerLook().a(this.target.locX, this.a.locY + (double) this.a.getHeadHeight(), this.target.locZ, (float) this.a.L(), (float) this.a.K());
//            this.a.getControllerLook().a(this.target, (float)(this.a.L() + 20), (float)this.a.K());
        if (this.a.h(this.target) < 6.25D) {
            this.a.getNavigation().q();
        } else {
            this.a.getNavigation().a(this.target, this.b);
        }

    }
}

File: Mobzy-API/src/main/java/com/offz/spigot/mobzy/pathfinders/PathfinderGoalTemptPitchLock.kt
package com.offz.spigot.mobzy.pathfinders

import com.offz.spigot.mobzy.mobs.CustomMob
import com.offz.spigot.mobzy.toNMS
import net.minecraft.server.v1_15_R1.EntityLiving
import org.bukkit.Material
import org.bukkit.craftbukkit.v1_15_R1.event.CraftEventFactory
import org.bukkit.entity.LivingEntity
import org.bukkit.event.entity.EntityTargetEvent

class PathfinderGoalTemptPitchLock(private val mob: CustomMob, private val targetItems: List<Material>) : com.offz.spigot.mobzy.pathfinders.PathfinderGoal() {
    private var target: LivingEntity? = null
    private val navigation = mob.navigation
    private val entity = mob.entity
    private var i: Int = 0

    override fun shouldExecute(): Boolean {
        /*return if (i > 0) {
            --i
            false
        } else {*/
        target = mob.findNearbyPlayer(10.0)?.living
        if (target == null) return false

        val equipment = target!!.equipment ?: return false

        return if (targetItems.any { it == equipment.itemInMainHand.type || it == equipment.itemInOffHand.type }) {
            //run the event
            val event = CraftEventFactory.callEntityTargetLivingEvent(entity, target?.toNMS() as EntityLiving, EntityTargetEvent.TargetReason.TEMPT)
            if (event.isCancelled) return false
            true
        } else false
    }

    override fun shouldTerminate(): Boolean = false

    override fun init() {
    }

    override fun reset() {
    }

    override fun execute() {
        mob.lookAtPitchLock(target!!)
        if (mob.distanceToEntity(target!!) < 6.25) navigation.moveToEntity(target!!)
    }
}